solr/core/src/test/org/apache/solr/handler/admin/ThreadDumpHandlerTest.java

```diff
@@ -214,16 +261,36 @@ public void doTestOwnableSync(final boolean checkWaitingThreadViaPolling) throws
           Thread.sleep(10); // 10ms at a time, at most 5 sec total
         }
         if (lock.hasQueuedThread(blockedT)) {
-          assertQ(req("qt", "/admin/threads", "indent", "true")
-                  // same lock owner 'ownerT'
-                  , "//lst[@name='thread'][str[@name='name'][.='test-thread-sync-lock-owner']]"
-                  + "                     [arr[@name='synchronizers-locked']/str[contains(.,'ReentrantLock')]]"
-                  // blocked thread 'blockedT', waiting on the lock
-                  , "//lst[@name='thread'][str[@name='name'][.='test-thread-sync-lock-blocked']]"
-                  + "                     [str[@name='state'][.='WAITING']]"
-                  + "                     [lst[@name='lock-waiting'][lst[@name='owner']/str[.='test-thread-sync-lock-owner']]]"
-                  );
-          
+          @SuppressWarnings({"unchecked"})
+          NamedList<Object> blockedThreads = (NamedList<Object>) readProperties()._get("system/threadDump", null);
+          found = false;
+          for (Map.Entry<String, Object> threadEntry : blockedThreads) {
+            @SuppressWarnings({"unchecked"})
+            NamedList<Object> thread = (NamedList<Object>) threadEntry.getValue();
+            // lock owner 'ownerT'
+            if (thread._getStr("name",  null).contains("test-thread-sync-lock-owner")) {
+              if (thread._get("synchronizers-locked", "").toString().contains("ReentrantLock")) {
+                found = true;
+                break;
+              }
+            }
+          }
```

Comment:
Can this logic be refactored into a helper method? It looks like it's been repeated several times already, looking for a thread with a given name and some other condition that could be expressed via a Predicate.